Cargo.lock

```diff
@@ -22,20 +22,29 @@ dependencies = [
 
 [[package]]
 name = "ahash"
-version = "0.7.6"
+version = "0.8.3"
```

Comment:
This is not related to this PR, but we may want to evaluate if we should keep this file (Cargo.lock) in the repository or if we add it to the .gitignore. Reference: https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries

Reply:
For now, I am fine with leaving it in, but something to consider for the future